components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -129,6 +129,15 @@ private OAuth2AuthorizeRespDTO validateAuthzRequest(OAuth2AuthorizeReqDTO authzR
             return authorizeRespDTO;
         }
 
+        List<String> allowedScope = OAuthServerConfiguration.getInstance().getAllowedScopes();
+        List<String> requestedAllowedScopes = new ArrayList<>();
+        String[] requestedScope = authzReqMsgCtx.getAuthorizationReqDTO().getScopes();
+        for (String scope : requestedScope) {
+            if (OAuth2Util.isAllowedScope(allowedScope, scope)) {
+                requestedAllowedScopes.add(scope);
+            }
+        }
+
```

Comment:
We need to remove these scopes from `authzReqMsgCtx` before engaging the  `validateScope` method.

Check the usage of `removeInternalScopes` method for reference.

Reply:
Some scopes which can come under allowed scopes like openId can used at the grant handler level as well, so we can not remove it from the requested scopes like internal scopes.  @malithie can confirm this, as we had a discussion on this previously as well.

Reply:
In the default scope validator we have specially handled[1] the `openid` scope so we don't need to add it to the allowed scopes. If we don't skip the allowed scopes from scope validators then there is a chance one of the scope validators will fail the request.

[1] - https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCScopeValidator.java#L208

Reply:
To address the requirement, the correct approach is to skip passing allowed scopes for further validations for downstream handlers.

To do that, what I suggested was to identify the impact of allowing scopes at the handler level, based on the fact that those are being utilized at a grant handler or lower levels for decisions.
In such a case, we need to take such validations to the handler level and fix the downstream handlers, and also capture the backward compatibility concerns (document) in case someone has extended in lower levels and have taken decisions.

Reply:
ack , addressed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1468/commits/712da68b56ac47cad8d15e773720e036846df432